runtime-modules/bounty/src/lib.rs

```diff
@@ -269,6 +285,9 @@ pub enum BountyStage {
     /// Bounty funding period expired with no contributions.
     FundingExpired,
 
+    /// Bounty cancelled in funding period expired with no contributions.
+    Cancelled,
```

Comment:
After reviewing the full pallet end to end, the complexity is just so high, I don't think we can justify keeping the `cancel_bounty` extrinsic or stage & milestone, when we also have `terminate_bounty`. They only have slightly different permissions, so I think we should just allow the the creator to call the `terminate_bounty` extrinsic only while in the funding period (even if there are contributions), and then we have covered most of the cancel use case with less complexity.